### PR TITLE
Feature/curator lo filter

### LINKED ISSUE #667: View and Filter Learning Objects in Collection Dashboard
As a curator/editor/admin I want to see and filter a list of all objects I have access to in the collection dashboard so that I can easily find a Learning Object.

### PR DESCRIPTION
**Pull request by jbarlow96**
As a curator/editor/admin I want to see and filter a list of all objects I have access to in the collection dashboard so that I can easily find a Learning Object.

closes #667

### CODE REVIEW

**Comment by jbarlow96** [comment]
screen shot 2019-03-06 at 6 01 32 pm
screen shot 2019-03-06 at 6 01 21 pm
screen shot 2019-03-06 at 6 02 02 pm

**Comment by jbarlow96** [comment]
Currently just fixing the indicator on the collection filter dropdown. I accidentally messed it up and am now just repairing it.

File: src/app/admin/admin.module.ts [hunk 1]
```diff
@@ -7,6 +7,14 @@ import { LearningObjectsComponent } from './pages/learning-objects/learning-obje
 import { AnalyticsComponent } from './pages/analytics/analytics.component';
 import { UsersComponent } from './pages/users/users.component';
 import { ContentWrapperComponent } from './components/content-wrapper/content-wrapper.component';
+import { FilterSearchComponent } from 'app/admin/components/filter-search/filter-search.component';
+import { FormsModule } from '@angular/forms';
```
Comment (nvisal1) [inline src/app/admin/admin.module.ts]: Are all of these imports necessary?

File: src/app/admin/admin.module.ts [hunk 2]
```diff
@@ -7,6 +7,14 @@ import { LearningObjectsComponent } from './pages/learning-objects/learning-obje
...
+import { FilterSearchService } from 'app/shared/filter-search.service';
+import { TooltipModule } from 'app/shared/tooltips/tip.module';
+import { ContextMenuModule } from 'app/shared/contextmenu/contextmenu.module';
+import { DashboardComponent } from 'app/onion/dashboard/dashboard.component';
+import { ModalModule } from 'app/shared/modals';
+import { PopupModule } from 'app/shared/popups/popup.module';
```
Comment (nvisal1) [inline src/app/admin/admin.module.ts]: Are all of these imports necessary?

File: src/app/admin/components/filter-search/filter-search.component.html [hunk 3]
```diff
@@ -0,0 +1,69 @@
+
+  <div class="list">
```
Comment (nvisal1) [inline src/app/admin/components/filter-search/filter-search.component.html]: I started following BEM in the Admin module. Change the selectors to follow BEM for consistency. 
Docs: http://getbem.com/

File: src/app/admin/components/filter-search/filter-search.component.html [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          <div *ngIf="filters.get('review')" class="indicator"></div>
+          <i class="far fa-sync"></i>Under Review
+        </li>
+        <li class="released" (click)="toggleStatusFilter('released')">
+          <div *ngIf="filters.get('released')" class="indicator"></div>
+          <i class="far fa-eye"></i>Released
+        </li>
+        <li class="denied" (click)="toggleStatusFilter('rejected')">
+          <div *ngIf="filters.get('rejected')" class="indicator"></div>
+          <i class="far fa-ban"></i>Denied
```
Comment (nvisal1) [inline src/app/admin/components/filter-search/filter-search.component.html]: Instead of hardcoding each list item, try using an ngFor.

File: src/app/admin/components/filter-search/filter-search.component.html [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        </li>
+        <li class="gencyber" (click)="toggleCollectionFilter('gencyber')">
+          <div *ngIf="filters.get('gencyber')" class="indicator"></div>GenCyber
+        </li>
+        <li class="intro_to_cyber" (click)="toggleCollectionFilter('intro_to_cyber')">
+          <div *ngIf="filters.get('intro_to_cyber')" class="indicator"></div>Intro to Cyber
+        </li>
+        <li class="nccp" (click)="toggleCollectionFilter('nccp')">
+          <div *ngIf="filters.get('nccp')" class="indicator"></div>NSA NCCP
+        </li>
```
Comment (nvisal1) [inline src/app/admin/components/filter-search/filter-search.component.html]: Instead of hardcoding each list item, try using an ngFor.
You should make a request to get collections before rendering the collection menu. This will make it easier to update the menu when we add more collections.

File: src/app/admin/components/filter-search/filter-search.component.ts [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+      private authService: AuthService,
+      private contextMenuService: ContextMenuService,
+      private router: Router) { }
+
+    ngOnInit() {
+      this.subToCollections();
+      this.findUserRestrictions();
+    }
+
+    private subToCollections(): void {
```
Comment (nvisal1) [inline src/app/admin/components/filter-search/filter-search.component.ts]: It looks like we are fetching collections here. We should use this information to populate the collection menu. Make sure that all functions in this file are being used and provide JSDoc for each.

File: src/app/admin/pages/learning-objects/learning-objects.component.ts [hunk 7]
```diff
@@ -73,4 +73,23 @@ export class LearningObjectsComponent {
   toggleStatus(status: string) {
     this.selectedStatus = status.toLowerCase();
   }
+  getFilteredLearningObjects(status: string, isStatus: boolean) {
+    console.log(status);
```
Comment (nvisal1) [inline src/app/admin/pages/learning-objects/learning-objects.component.ts]: Remove dev log

File: src/app/shared/filter-search.service.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  private statusFilter$: BehaviorSubject<string[]> = new BehaviorSubject<string[]>(
+    []
+  );
+
+   private selectedStatusFilter$: BehaviorSubject<string[]> = new BehaviorSubject<
+  string[]
+  >(null);
+  constructor() {}
+
+  public getFilters(): BehaviorSubject<Filter[]> {
```
Comment (nvisal1) [inline src/app/shared/filter-search.service.ts]: Make sure all of these copied functions are being used and provide JSDoc for each.

File: src/app/admin/admin.module.ts (hunk 1 above)
Comment (DonSeannelly) [reply src/app/admin/admin.module.ts]: @nvisal1 can you elaborate? They are all being used within the module, do you think they should be located at a different point in the module hierarchy?

File: src/app/admin/components/filter-search/filter-search.component.html (hunk 4 above)
Comment (DonSeannelly) [reply src/app/admin/components/filter-search/filter-search.component.html]: @nrwinner what are your thoughts on this? It will require the component class to have the icon class names built in. I see no architectural issue with it since it is still the component. Thoughts?

File: src/app/shared/filter-search.service.ts (hunk 8 above)
Comment (DonSeannelly) [reply src/app/shared/filter-search.service.ts]: This entire service is not being used anywhere. Can we remove it?

@gshaw1997 @nrwinner maybe you have context?

File: src/app/admin/components/filter-search/filter-search.component.html (hunk 4 above)
Comment (nrwinner) [reply src/app/admin/components/filter-search/filter-search.component.html]: @DonSeannelly @nvisal1 I'm on board with the introduction of ngFor here. I agree that we don't break any architectural by giving the component class names for icons.

File: src/app/shared/filter-search.service.ts (hunk 8 above)
Comment (nrwinner) [reply src/app/shared/filter-search.service.ts]: I don't have any context, I don't think I've ever seen that file. Could it be a remnant of the old admin dashboard @gshaw1997 ? Regardless, its purpose appears to be to allow the sharing of filters between components. This is an anti pattern considering we should be segregating filters to the components they apply to, I'm game to remove it.

File: src/app/admin/components/filter-search/filter-search.component.scss [hunk 9]
```diff
@@ -0,0 +1,403 @@
+@import '~_vars.scss';
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.scss]: Is this the entirety of the styling of CLARK's dashboard? Why is it here?

File: src/app/admin/components/filter-search/filter-search.component.html [hunk 10]
```diff
@@ -0,0 +1,42 @@
+
+  <div class="filter">
+    <div #listInner class="filter__container">
+      <div class="container__list-top">
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.html]: Why are we nesting so many levels here?

File: src/app/admin/components/filter-search/filter-search.component.scss [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+
+        .indicator {
+          background: white;
+        }
+      }
+    }
+  }
+}
+
+.loading-template {
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.scss]: Why is this loading template here? To my knowledge the filters menu doesn't loading anything asynchronously?

File: src/app/admin/components/filter-search/filter-search.component.scss [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+  padding: 20px;
+
+  .svg-inline--fa {
+    color: $bright-blue;
+    font-size: 30px;
+    margin-bottom: 20px;
+  }
+}
+
+.empty-template {
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.scss]: Why is this here? Are we using this element anywhere?

File: src/app/admin/components/filter-search/filter-search.component.ts [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+  private subscriptions: Subscription[] = [];
+  collections: { abvName: string, name: string }[] = [];
+  isCollectionRestricted = false;
+  filtersModified$: Subject<void> = new Subject();
+  selectedCollection: string;
+  // collection
+  // filters applied to dashboard objects (status filters)
+  filters: Map<string, boolean> = new Map();
+
+  statuses = [
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.ts]: Why not use the variables in clark-entity here?

File: src/app/admin/components/filter-search/filter-search.component.scss [hunk 14]
```diff
@@ -0,0 +1,317 @@
...
+  position: absolute;
+  left: 0;
+  right: 0;
+  top: 0;
+  bottom: 0;
+
+  color: rgba(red($dark-text), green($dark-text), blue($dark-text), 0.6);
+  font-size: 16px;
+
+  .top {
```
Comment (nrwinner) [inline src/app/admin/components/filter-search/filter-search.component.scss]: What does this style?

**Comment by nrwinner** [comment]
Almost there! Just a few small things.
